version: '3.8'

services:
  postgresql:
    container_name: lavagna-db
    image: 10.26.0.176:5000/postgres:9.4
    restart: always
    environment:
      POSTGRES_DB: "${DB_NAME}"
      POSTGRES_USER: '${DB_USER}'
      POSTGRES_PASSWORD: $DB_PASSWORD
    ports:
      - "5432:5432"
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./postgres-datadog-db.conf.sql:/docker-entrypoint-initdb.d/postgres-datadog-db.conf.sql
    networks:
      - lavagna

  app:
    container_name: lavagna-app
    build:
      context: .
      dockerfile: Dockerfile.deploy
      args:
        DB_DIALECT: ${DB_DIALECT}
        DB_URL: "${DB_URL}"
        DB_USERNAME: "${DB_USER} "
        DB_PASSWORD: "${DB_PASSWORD}"
        SPRING_PROFILES_ACTIVE: "dev"
    depends_on:
      - postgresql
    ports:
      - "8080:8080"
    networks:
      - lavagna

  datadog:
    build: 
      context: ./datadog
      dockerfile: Dockerfile
      args:
        - POSTGRES_HOST=lavagna-app
        - POSTGRES_PORT=5432
        - POSTGRES_USER=datadog
        - POSTGRES_PASSWORD=datadog
    pid: host
    environment:
      - DD_SITE=${DATADOG_SITE}
      - DD_API_KEY=${DATADOG_API_KEY}
      - DD_LOGS_ENABLED=true
    volumes:
     - /var/run/docker.sock:/var/run/docker.sock
     - /proc/:/host/proc/:ro
     - /sys/fs/cgroup:/host/sys/fs/cgroup:ro
     - /var/lib/docker/containers:/var/lib/docker/containers:ro
    networks:
      - lavagna

volumes:
  postgres-data:
    driver: local

networks:
  lavagna:
    driver: bridge